algo

```diff
@@ -1,4 +1,4 @@
-#!/bin/sh
+#!/bin/bash
```

Comment:
On Ubuntu, /bin/sh is dash instead of bash. And that uses a different version of the read command. You get this message: "./algo: 7: read: Illegal option -s" on Ubuntu 16.04